win32ss/gdi/ntgdi/d3dkmt.c

```diff
@@ -0,0 +1,712 @@
+/*
+ * PROJECT:     ReactOS Display Driver Model
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     D3DKMT dxgkrnl callbacks
+ * COPYRIGHT:   Copyright 2023 Justin Miller <justin.miller@reactos.org>
+ */
+
+#include <win32k.h>
+#include <reactos/rddm/rxgkinterface.h>
+#include <debug.h>
+
+/*
+ * It looks like Windows saves all the function pointers globally inside win32k.
+ * Instead, we're going to keep it static to this file and keep it organized in struct
+ * we obtained with the IOCTL.
+ */
+static REACTOS_WIN32K_DXGKRNL_INTERFACE DxgAdapterCallbacks = {0};
+
+/*
+ * This looks like it's done inside DxDdStartupDxGraphics, but I'd rather keep this organized.
+ * Dxg gets start inevitably anyway it seems at least on vista.
+ */
+VOID
+DxStartupDxgkInt()
+{
+    DPRINT("DxStartupDxgkInt: Entry\n");
+    //TODO: Let DxgKrnl know it's time to start all adapters, and obtain the win32k<->dxgkrnl interface via an IOCTRL.
+}
+
+BOOLEAN
+APIENTRY
+NtGdiDdDDICheckExclusiveOwnership(VOID)
+{
+    // We don't support DWM at this time, excusive ownership is always false.
+    return FALSE;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDIGetProcessSchedulingPriorityClass(_In_  HANDLE                                unnamedParam1,
```

Comment:
Both here and in the prototypes: I am not a fan of the `unnamedParam1`, but I understand why you went that way.

Reply:
Also good to note it's like this on MSDN
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/d3dkmthk/nf-d3dkmthk-d3dkmtgetprocessschedulingpriorityclass

Reply:
yeah they broke a bunch like this of things when moving to markdown files.